Mailer/app/api/v1/Handlers/Mailers/MailLogsHandler.php

```diff
@@ -2,54 +2,55 @@
 
 namespace Remp\MailerModule\Api\v1\Handlers\Mailers;
 
+use Nette\Utils\Arrays;
 use Nette\Utils\Json;
-use Nette\Utils\JsonException;
-use Remp\MailerModule\Repository\ListsRepository;
+use Remp\MailerModule\ActiveRow;
+use Remp\MailerModule\Api\Validation\InputValidatorTrait;
 use Remp\MailerModule\Repository\LogsRepository;
-use Remp\MailerModule\Repository\MailTypesRepository;
 use Tomaj\NetteApi\Handlers\BaseHandler;
-use Tomaj\NetteApi\Params\InputParam;
 use Tomaj\NetteApi\Response\JsonApiResponse;
-use Tracy\Debugger;
 
 class MailLogsHandler extends BaseHandler
 {
     private $logsRepository;
 
+    use InputValidatorTrait;
+
     public function __construct(
         LogsRepository $logsRepository
     ) {
         parent::__construct();
         $this->logsRepository = $logsRepository;
     }
 
-    public function params()
-    {
-        return [
-            new InputParam(InputParam::TYPE_POST, 'where', InputParam::OPTIONAL),
-            new InputParam(InputParam::TYPE_POST, 'limit', InputParam::OPTIONAL),
-        ];
-    }
-
-    public function handle($params)
+    public function handle($payload)
     {
-        $output = $this->logsRepository->all();
+        $request = file_get_contents("php://input");
+        $payload = $this->validateInput($request, 'logs.schema.json');
 
-        if (!empty($params['where'])){
-            $output->where(Json::decode($params['where'], Json::FORCE_ARRAY));
+        if ($this->hasErrors()){
+            return $payload;
         }
 
-        if (!empty($params['limit'])){
+        $output = $this->logsRepository->all()->where($payload['filter'] . ' IS NOT NULL');
```

Comment:
- Never build database condition by concatenating strings. Never.
- What if someone wants to filter multiple types of emails?

Reply:
I assumed that only one type of email can be filtered in one result.

Reply:
It will be from the point where you'll be calling it from, but this API should be more universal. We agreed on the most simplest 1:1 implementation in the beginning (which you did fine), but sometimes we'll need to revisit these and make these APIs to look more like APIs. This is one of them.

This is basically API for listing any kind of mail logs, so filtering based on various parameters (`mail_template_id`, `email`) should be considered. Also, the date fields should be filterable based on their values (such as "sent after DATE_A" or "delivered after DATE_A and before DATE_B").

Could you prepare schema, which would still satisfy CRM needs and also would allow any other clients to get the data various ways? We could iterate on that.